src/python_testing/TC_SOIL_2_2.py

```diff
@@ -0,0 +1,108 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --endpoint 1
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+from random import randrange
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.testing.matter_testing import MatterBaseTest, TestStep, default_matter_test_main, has_cluster, run_if_endpoint_matches
+from mobly import asserts
+
+
+class TC_SOIL_2_2(MatterBaseTest):
+    async def read_soil_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.SoilMeasurement
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_SOIL_2_2(self) -> str:
+        return "[TC-SOIL-2.2] Primary functionality with DUT as Server"
+
+    def steps_TC_SOIL_2_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read SoilMoistureMeasurementLimits attribute, save the MinMeasuredValue field as min_bound and save the MinMeasuredValue field as max_bound"),
+            TestStep(3, "Read SoilMoistureMeasuredValue attribute and save the value as measurement"),
+            TestStep(4, "Perform action to change the moisture of the measured medium"),
+            TestStep(5, "After a few seconds, read SoilMoistureMeasuredValue attribute"),
+        ]
+        return steps
+
+    def pics_TC_SOIL_2_2(self) -> list[str]:
+        pics = [
+            "SOIL.S",
+        ]
+        return pics
+
+    @run_if_endpoint_matches(has_cluster(Clusters.SoilMeasurement))
```

Comment:
In the logs I see:

```
04-30 10:54:55.789 INFO Test is not applicable to this endpoint - skipping test
```

which may be ok because not all endpoints have these clusters, however have we verified that all-clusters has at least one cluster that matches?

maybe overall issue: for CI tests we should maybe ensure that at least one test is actually executed, not sure if we have such a thing.

Reply:
#38687 created to figure it out in general.

For this particular case though, please add to the testing summary how you ensured that the apps you use (all clusters) does indeed exercise what we want.

Reply:
Hmm, interesting. I added the run_if_endpoint_matches due to a suggestion from Cecille, but I figured it should have executed the test regardless 🤔

Reply:
Arh, hold on. I guess this is because the server implies has not yet landed. So from the perspective of this PR, the cluster is not present, which is why it skips. At least that is my assumption.